src/minmax.rs

```diff
@@ -0,0 +1,90 @@
+use rustc::lint::{Context, LintPass, LintArray};
+use rustc_front::hir::*;
+use syntax::codemap::Spanned;
+use syntax::ptr::P;
+use std::cmp::PartialOrd;
+use std::cmp::Ordering::*;
+
+use consts::{Constant, constant};
+use utils::{match_path, span_lint};
+use self::MinMax::{Min, Max};
+
+declare_lint!(pub MIN_MAX, Deny,
```

Comment:
Um, not sure. I'm actually uneasy about bad_bit_mask being Deny (for the same reasons), though zero_width_space and cmp_nan seem fine.